lib/escalate/mixin.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Escalate
+  module Mixin
+    class << self
+      def included(base)
+        raise 'instead of `include ExEscalator::Mixin`, you should `include ExEscalator.mixin`'
```

Comment:
ExEscalator -> Escalate :)

I wonder about hiding this even more. We could name the module `__Mixin` and rename the `mixin` method to `Mixin`. Ruby actually allows methods to have leading uppercase (so they look like a constant). `Integer` and `Array` are examples.
But to reach that method, you'll get an error if you try `Escalate::Mixin`. You have to do `Escalate.Mixin`. Hmm, that might be maddening in itself.

Reply:
I don't think we need to hide this more. It's good to have that knowledge that there is a mixin, and this can't be included (or even if it was, it wouldn't do anything) and we direct that there is method to use to get this mixed in properly. The other approach would be to make this includable, and to make it so you only need to use `Escalate.mixin` iff you need to pass a logger block.  This would make the 90% case `include Escalate::Mixin` and the 10% case `Escalate.mixin { some_other_logger }`.  Thoughts on that?

Reply:
I'd rather not encourage people to use the `::Mixin` at all. That's why I'm tempted to name that `_Mixin` or `MixinInternal` or something like that.

Reply:
I went ahead and removed the temptation entirely by moving the entire mixin module into the `Module.new` in the `Escalate.mixin` method. That way there's nothing in `Escalate::Mixin` or any other statically defined modules to get tripped up by.  The only option is to use `Escalate.mixin`

e6733a0d70884986eee5717b3040076b15b90f90

Reply:
👍 Removing temptation--always a good idea!